### PR TITLE
first pass at setup for GH Actions to build / deploy

### PR DESCRIPTION
**Pull request by thorwolpert**
This should have all the templates and files to do build/deploys for the ppr-api

Note: added a new vault for ppr

### CODE REVIEW

File: ppr-api/Makefile [hunk 1]
```diff
@@ -84,6 +84,7 @@ cd: build ## CD flow
 
 build: ## Build the docker container
 	docker build -t $(DOCKER_NAME) \
+	--platform linux/amd64 \
 	--build-arg VCS_REF=$(shell git rev-parse --short HEAD) \
 	--build-arg BUILD_DATE=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ") \
 	.
```
Comment (pwei1018) [inline ppr-api/Makefile]: The CD part need to add the "update-env" step like the following.
#################################################################################
# COMMANDS - CD
# expects the terminal to be openshift login
# expects export OPENSHIFT_DOCKER_REGISTRY=""
# expects export OPENSHIFT_SA_NAME="$(oc whoami)"
# expects export OPENSHIFT_SA_TOKEN="$(oc whoami -t)"
# expects export OPENSHIFT_REPOSITORY=""
# expects export TAG_NAME="dev/test"
# expects export OPS_REPOSITORY=""                                                        #
#################################################################################
cd: build update-env tag ## CD flow

build: ## Build the docker container
	docker build . -t $(DOCKER_NAME) \
		--build-arg VCS_REF=$(shell git rev-parse --short HEAD) \
		--build-arg BUILD_DATE=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ") \

build-nc: ## Build the docker container without caching
	docker build --no-cache -t $(DOCKER_NAME) .

REGISTRY_IMAGE=$(OPENSHIFT_DOCKER_REGISTRY)/$(OPENSHIFT_REPOSITORY)-tools/$(DOCKER_NAME)
push: #build ## Push the docker container to the registry & tag latest
	@echo "$(OPENSHIFT_SA_TOKEN)" | docker login $(OPENSHIFT_DOCKER_REGISTRY) -u $(OPENSHIFT_SA_NAME) --password-stdin ;\
    docker tag $(DOCKER_NAME) $(REGISTRY_IMAGE):latest ;\
    docker push $(REGISTRY_IMAGE):latest

VAULTS=`cat devops/vaults.json`
update-env: ## Update env from 1pass
	oc -n "$(OPS_REPOSITORY)-tools" exec dc/bcros-cli -- ./scripts/1pass.sh \
		-m "secret" \
		-e "$(TAG_NAME)" \
		-a "$(DOCKER_NAME)-$(TAG_NAME)" \
		-n "$(OPENSHIFT_REPOSITORY)-$(TAG_NAME)" \
		-v "$(VAULTS)" \
		-r "true" \
		-f "false"

tag: push ## tag image
	oc -n "$(OPENSHIFT_REPOSITORY)-tools" tag $(DOCKER_NAME):latest $(DOCKER_NAME):$(TAG_NAME)

File: ppr-api/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,10 @@
 FROM python:3.8.6-buster
 USER root
```
Comment (pwei1018) [inline ppr-api/Dockerfile:2]: Add labels here?

Comment (thorwolpert) [reply ppr-api/Dockerfile:2]: must have copy/pasted over it.
they should be there now.

File: ppr-api/Makefile (hunk 1 above)
Comment (thorwolpert) [reply ppr-api/Makefile]: updated to the current CD Flow

File: ppr-api/k8s/templates/nsp.yaml [hunk 3]
```diff
@@ -0,0 +1,177 @@
+# Copyright 2021 The Province of British Columbia
```
Comment (pwei1018) [inline ppr-api/k8s/templates/nsp.yaml:1]: Don't need this one any more.

File: .github/workflows/ppr-api-cd.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    if: github.repository == 'bcgov/lear'
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set env by input
+        run: |
+          echo "TAG_NAME=${{ github.event.inputs.environement }}" >> $GITHUB_ENV
+
+      - name: Set env by master branch
+        if: env.TAG_NAME == '' && github.ref == 'refs/heads/master'
```
Comment (pwei1018) [inline .github/workflows/ppr-api-cd.yml]: switch to main.

File: ppr-api/k8s/templates/bc.yaml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    your just experimenting.
+  displayName: Git Repo URL
+  name: GIT_REPO_URL
+  required: true
+  value: https://github.com/bcgov/ppr.git
+- description: The git reference or branch.
+  displayName: Git Reference
+  name: GIT_REF
+  required: true
+  value: master
```
Comment (pwei1018) [inline ppr-api/k8s/templates/bc.yaml]: Switch to main.

**Review by pwei1018 — APPROVED** [review]

**Pull request merged**